Evolutions throws NullPointerException with MySQL, useLocks = true and cacheResultSetMetadata = true

### Play Version
2.7.3

### API
Both

### Operating System
Linux autumn 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux

### JDK

openjdk version "1.8.0_222"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_222-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.222-b10, mixed mode)

### Library Dependencies
"mysql" % "mysql-connector-java" % "8.0.18",

### Expected Behavior

1. App starts with db migration applied by evolutions

### Actual Behavior

1. App throws the following exception:
```
Caused by: java.lang.NullPointerException: null
	at com.mysql.cj.protocol.a.result.NativeResultset.initRowsWithMetadata(NativeResultset.java:123)
	at com.mysql.cj.jdbc.result.ResultSetImpl.initializeWithMetadata(ResultSetImpl.java:308)
	at com.mysql.cj.jdbc.ConnectionImpl.initializeResultsMetadataFromCache(ConnectionImpl.java:2343)
	at com.mysql.cj.jdbc.StatementImpl.executeQuery(StatementImpl.java:1195)
	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
	at play.api.db.evolutions.ApplicationEvolutions$.$anonfun$lock$1(ApplicationEvolutions.scala:274)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at play.api.db.evolutions.ApplicationEvolutions$.lock(ApplicationEvolutions.scala:274)
	at play.api.db.evolutions.ApplicationEvolutions$.withLock(ApplicationEvolutions.scala:225)
	at play.api.db.evolutions.ApplicationEvolutions$.runEvolutions(ApplicationEvolutions.scala:202)
	at play.api.db.evolutions.ApplicationEvolutions.$anonfun$start$1(ApplicationEvolutions.scala:68)
	at play.api.db.evolutions.ApplicationEvolutions.$anonfun$start$1$adapted(ApplicationEvolutions.scala:63)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at play.api.db.evolutions.ApplicationEvolutions.start(ApplicationEvolutions.scala:63)
	at play.api.db.evolutions.ApplicationEvolutions.<init>(ApplicationEvolutions.scala:109)
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get$lzycompute(EvolutionsModule.scala:68)
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:61)
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:51)
```

### Reproducible Test Case
Have the following in `application.conf`:
```
play.evolutions.useLocks = true
db.default.hikaricp.dataSource.cacheResultSetMetadata = true
```

### Suggested fix
This may be a MySQL connector/j bug, but using `Statement.execute` instead of `Statement.executeQuery` for `set innodb_lock_wait_timeout = 1` will avoid this issue. The ResultSet of that query and the following "select `lock` ... for update" are not used anyway.

--- a/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/ApplicationEvolutions.scala
+++ b/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/ApplicationEvolutions.scala
@@ -268,7 +268,7 @@ private object ApplicationEvolutions {
       case _               => lockPlayEvolutionsLockSqls
     }
     try {
-      for (script <- lockScripts) s.executeQuery(applySchema(script, dbConfig.schema))
+      for (script <- lockScripts) s.execute(applySchema(script, dbConfig.schema))
     } catch {
       case e: SQLException =>
         if (attempts == 0) throw e

